remove unverified certs once an email is verified

seanmonstar opened the pull request:
fixes #2924

@ozten can you make sure what I'm doing doesn't seem absoutely batty?

seanmonstar committed:
remove unverified certs once an email is verified

fixes #2924

ozten commented on resources/static/common/js/user.js:
```diff
@@ -260,20 +260,28 @@ BrowserID.User = (function() {
+      if (info.state === "unverified") {
+        email_obj.unverified = true;
+      } else if (email_obj.unverified) {
+        delete email_obj.unverified;
```
Nice.

ozten commented on resources/static/dialog/js/misc/state.js:
```diff
@@ -424,6 +424,12 @@ BrowserID.State = (function() {
       else {
+        // make sure an unverified-certs are removed
+        if (record.unverified && info.state !== 'unverified') {
+          storage.invalidateEmail(email);
```
Very cool.

ozten commented:
Frontend Tests failing with

```
request for 'get /wsapi/address_info?email=testemail%40testemail.com&issuer=default valid' is not mocked in resources/static/test/cases/xhr.js
```

Restarting Travis Build.

seanmonstar committed:
update user.js tests to use mocked email

changes to syncEmailKeypair to now check addressInfo
were calling an xhr with the email of testemail@testemail.com.
We didn't have those XHRs mocked, so everything blew up.

Switched all the usages to the mocked email address of
testuser@testuser.com

seanmonstar commented:
@ozten fixed up the tests!
